src/lib/y2storage/filesystems/blk_filesystem.rb

```diff
@@ -0,0 +1,87 @@
+require "y2storage/storage_class_wrapper"
+require "y2storage/filesystems/base"
+
+module Y2Storage
+  module Filesystems
+    # A local filesystem.
+    #
+    # This a wrapper for Storage::BlkFilesystem
+    class BlkFilesystem < Base
+      wrap_class Storage::BlkFilesystem
+
+      storage_class_forward :all, as: "Filesystems::BlkFilesystem"
+
+      storage_forward :supports_label?, to: :supports_label
+      storage_forward :max_labelsize
+      storage_forward :label
+      storage_forward :label=
+
+      storage_forward :supports_uuid?, to: :supports_uuid
+      storage_forward :uuid
+      storage_forward :uuid=
+
+      storage_forward :mkfs_options
+      storage_forward :mkfs_options=
+      storage_forward :tune_options
+      storage_forward :tune_options=
+      storage_forward :detect_content_info
+
+      # @!method blk_devices
+      #   Formatted block devices. It returns the block devices directly hosting
+      #   the filesystem. That is, for encrypted filesystems it returns the
+      #   encryption devices.
+      #
+      #   In most cases, this collection will contain just one element, since
+      #   most filesystems sit on top of just one block device.
+      #   But that's not necessarily true for Btrfs, see
+      #   https://btrfs.wiki.kernel.org/index.php/Using_Btrfs_with_Multiple_Devices
+      #
+      #   @return [Array<BlkDevice>]
+      storage_forward :blk_devices, as: "BlkDevice"
+
+      # Raw (non encrypted) version of the formatted devices. If the filesystem
+      # is not encrypted, it returns the same collection that #blk_devices,
+      # otherwise it returns the original devices instead of the encryption
+      # ones.
+      #
+      # @return [Array<BlkDevice>]
+      def plain_blk_devices
+        blk_devices.map { |dev| dev.is_a?(Encryption) ? dev.blk_device : dev }
```

Comment:
well, this is a bit confusing for me. BlkDevice have method encrypted? and plain_blk_device. So why it is not used? Even now when I know code it is not clear for me why it do not look like `blk_devices.map(&:plain_blk_device)`